keps/sig-scheduling/3101-pod-to-node-assignment/README.md

```diff
@@ -0,0 +1,936 @@
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This Kep proposes adding a feature (`PodsPerNodeAssignmentPolicy`) that provides
+users with an explicit API and scheduler support for 1:1 pod to node assignment.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Customers trying to deploy batch workloads on Kubernetes have filed multiple
+requests (See [1](https://github.com/kubernetes/kubernetes/issues/68827),
+[2](https://github.com/kubernetes/kubernetes/issues/105287) for details) for 1:1
+pod-to-node assignment. The most common use case for enforcing 1:1 pod to node
+assignment is HPC workloads (MPI and Spark).
+
+Currently there are two approaches to achieve this:
+
+-   Set resource requests to something high enough to prevent scheduling any
+    other pods but daemonset and static pods. This is tricky to do and error
+    prone, especially for smaller nodes. This will not be possible if node size
+    is not known in advance.
+-   Use pod anti affinity. Users will need to define the same set of labels on
+    all pods that needs to be repelled, whether or not they actually request 1:1
+    pod to node assignment, it is also expensive to calculate especially for
+    cluster autoscaler.
+
+A simple and native way to force 1:1 pod-to-node assignment is therefore needed.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Allow users to force 1:1 pod-to-node assignment for a given pod
+
+### Non-Goals
+
+-   Find the best fit node for a given pod
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Add a new pod-level API that allows users to specify how many "worker" pods are
```

Comment:
I agree this can be rephrased, but the new wording might still be confusing as the API could allow coexistence with other pods. Would it be better stated as
```
Add a new pod-level API that allows users to enforce how a pod can coexist with other pods when scheduled on a node. Available policies are:
- `DaemonsetAndStaticPods`: pod only tolerates daemonset and static pods to be running on the node.
- `Any`: pod tolerates any pod type to be running on the node (default)
```

Reply:
SGTM.